tests/unit/s2n_quic_support_io_test.c

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_quic_support.h"
+
+#include "tls/s2n_connection.h"
+#include "tls/s2n_handshake.h"
+#include "tls/s2n_tls.h"
+#include "utils/s2n_mem.h"
+
+/* We need access to some io logic */
+#include "tls/s2n_handshake_io.c"
+
+static const uint8_t TEST_DATA[] = "test";
+static const size_t TEST_DATA_SIZE = sizeof(TEST_DATA);
+
+struct s2n_stuffer input_stuffer, output_stuffer;
+static int s2n_setup_conn(struct s2n_connection *conn)
```

Comment:
Would it be a good idea for new tests to be using the `S2N_RESULT` return type instead? It should make the migration easier over time :)